go/pkg/client/tree.go

```diff
@@ -321,10 +321,9 @@ func buildTree(files map[string]*fileSysNode) (*treeNode, error) {
 			if node.dirs == nil {
```

Comment:
Not related to this change (and don't have to fix it): it looks like buildTree() is supposed to work with files and empty virtual dir is more of a special case handling that it does? If so it maybe more reasonable to have this done in loadFiles()?

Reply:
I think buildTree is a reasonable place since it builds the tree of directories and files, and is the best place to know about collisions. loadFiles doesn't iterate over path segments for instance, so it won't have visibility into collections in ancestor directories.

Reply:
Ah I see what you're saying. I think its fair to rethink the interface of this function to not give that impression, the core of it is to construct a tree of directories and files. It should take in leaf nodes, whether directories or files, and build a tree out of them. Perhaps we can address this when we refactor the whole virtual inputs area.

Reply:
sgtm!